src/server/transaction.cc

```diff
@@ -411,17 +414,23 @@ void Transaction::StartMultiGlobal(DbIndex dbid) {
   ScheduleInternal();
 }
 
-void Transaction::StartMultiLockedAhead(DbIndex dbid, CmdArgList keys) {
+void Transaction::StartMultiLockedAhead(DbIndex dbid, CmdArgVec keys) {
   DVLOG(1) << "StartMultiLockedAhead on " << keys.size() << " keys";
 
   DCHECK(multi_);
   DCHECK(shard_data_.empty());  // Make sure default InitByArgs didn't run.
 
   multi_->mode = LOCK_AHEAD;
-  InitBase(dbid, keys);
+  multi_->lock_mode = LockMode();
+
+  CopyMultiKeys(&keys);  // Filter uniques and normalize
+
+  InitBase(dbid, absl::MakeSpan(keys));
   InitByKeys(KeyIndex::Range(0, keys.size()));
 
   ScheduleInternal();
+
+  full_args_ = {nullptr, 0};  // Was pointer to temporary
```

Comment:
where it was pointer to temporary? 
maybe worth resetting it there?

Reply:
What do you mean by resetting? It was a pointer to an outer scope temporary, now it's a pointer to an inner scope temporary, I just wanted to make it explicit

Reply:
You do not use this variable here so this statement is not immediately clear and the comment 
is not clear to me :)
I now inspected the code and understood now that you refer to keys.

Maybe better:
```suggestion
  full_args_ = {nullptr, 0};  // InitBase set it to temporary keys, now we reset it.
```

Another question - we use `full_args_` for replication. How does it work if you reset it here?

Reply:
> Another question - we use full_args_ for replication. How does it work if you reset it here?

We use it for replicating _commands after hops_, not the initialization step. Next MultiSwitchCmd() will reset it to a proper command slice and that will be picked up by hops.